### PR TITLE
Docker Desktop for Linux support

### PR DESCRIPTION
**Pull request by nicksieger**
Taking a stab at DD for linux. I don't have a working DD for linux I can use yet, so I'd appreciate anyone who could build and try this!

- chore: changes from `make fmt`
- cluster: add docker desktop for linux support

### CODE REVIEW

File: pkg/cluster/admin_docker_desktop.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (a *dockerDesktopAdmin) Delete(ctx context.Context, config *api.Cluster) er
 	if !isLocalDockerEngine {
 		return fmt.Errorf("docker-desktop cannot be deleted from a remote DOCKER_HOST: %s", a.host)
 	}
-	if a.os != "darwin" && a.os != "windows" {
+	if a.os != "darwin" && a.os != "windows" && a.os != "linux" {
```
Comment (nicksieger) [inline pkg/cluster/admin_docker_desktop.go:53]: Can we delete this whole conditional?

File: pkg/cluster/docker_desktop_dial.go [hunk 2]
```diff
@@ -18,17 +18,26 @@ func dockerDesktopSocketPaths() ([]string, error) {
...
+			// Older versions of docker desktop use this socket.
+			filepath.Join(homedir, "Library/Containers/com.docker.docker/Data/gui-api.sock"),
+
+			// Newer versions of docker desktop use this socket.
+			filepath.Join(homedir, "Library/Containers/com.docker.docker/Data/backend.native.sock"),
+		}, nil
+	case "linux":
+		return []string{
+			// Docker Desktop for Linux
+			filepath.Join(homedir, ".docker/desktop/backend.native.sock"),
```
Comment (nicksieger) [inline pkg/cluster/docker_desktop_dial.go:33]: I was able to verify this location works in a VM, I just can't get DD4L fully started in a VM yet

**Review by milas — APPROVED** [review]

File: pkg/cluster/docker_desktop_dial.go (hunk 2 above)
Comment (nicksieger) [reply pkg/cluster/docker_desktop_dial.go]: Also, I'm going to just switch these to `backend.sock` since the native socket is going away.

File: pkg/cluster/docker_desktop_dial.go [hunk 3]
```diff
@@ -18,17 +18,26 @@ func dockerDesktopSocketPaths() ([]string, error) {
...
-		filepath.Join(homedir, "Library/Containers/com.docker.docker/Data/backend.native.sock"),
-	}, nil
+	switch runtime.GOOS {
+	case "darwin":
+		return []string{
+			// Older versions of docker desktop use this socket.
+			filepath.Join(homedir, "Library/Containers/com.docker.docker/Data/gui-api.sock"),
+
+			// Newer versions of docker desktop use this socket.
+			filepath.Join(homedir, "Library/Containers/com.docker.docker/Data/backend.sock"),
```
Comment (nicks) [inline pkg/cluster/docker_desktop_dial.go:28]: are you sure this works? i thought this socket used a different protocol

File: pkg/cluster/docker_desktop_dial_windows.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 func dockerDesktopSocketPaths() ([]string, error) {
 	return []string{
 		`\\.\pipe\dockerWebApiServer`,
-		`\\.\pipe\dockerBackendNativeApiServer`,
+		`\\.\pipe\dockerBackendApiServer`,
```
Comment (nicks) [inline pkg/cluster/docker_desktop_dial_windows.go:17]: ditto, i don't think this will work

**Comment by nicks** [comment]
ya, i have a linux box, lemme try to run this

File: pkg/docker/docker.go [hunk 5]
```diff
@@ -32,7 +32,9 @@ func IsLocalDockerEngineHost(dockerHost string) bool {
 		// see:
 		// https://github.com/tilt-dev/ctlptl/issues/196
 		// https://docs.docker.com/desktop/faqs/#how-do-i-connect-to-the-remote-docker-engine-api
-		return strings.Contains(dockerHost, "/var/run/docker.sock")
+		return strings.Contains(dockerHost, "/var/run/docker.sock") ||
+			// Docker Desktop for Linux - socket is in ~/.docker/desktop/docker.sock
+			strings.HasSuffix(dockerHost, "/.docker/desktop/docker.sock")
```
Comment (nicks) [inline pkg/docker/docker.go:37]: can you split out this fix into a separate PR? i think that would be worthwhile

Comment (nicks) [reply pkg/docker/docker.go:37]: (i wonder if we should have a library for docker socket cap detection, like we do with https://github.com/tilt-dev/clusterid)

Comment (nicksieger) [reply pkg/docker/docker.go:37]: Ok, so just the bits required for linux support separated from the backend socket changes and the formatting?

Comment (nicks) [reply pkg/docker/docker.go:37]: ya

**Review by nicks — APPROVED** [review]
now getting:

```
$ ctlptl delete -f examples/docker-desktop.yaml 
reset docker-desktop kubernetes: status code 404
```

but a lot better! maybe let's file that as a separate issue and merge this as-is
